hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/AvroSchemaEvolutionUtils.java

```diff
@@ -113,6 +120,21 @@ public static InternalSchema reconcileSchema(Schema incomingSchema, InternalSche
       typeChange.updateColumnType(col, inComingInternalSchema.findType(col));
     });
 
+    // mark columns missing from incoming schema as nullable
+    Set<String> visited = new HashSet<>();
+    diffFromOldSchema.stream()
+        // ignore meta fields
+        .filter(col -> !META_FIELD_NAMES.contains(col))
+        .sorted()
+        .forEach(col -> {
+          // if parent is marked as nullable, only update the parent and not all the missing children field
+          String parent = TableChangesHelper.getParentName(col);
+          if (!visited.contains(parent)) {
+            typeChange.updateColumnNullability(col, true);
+          }
+          visited.add(col);
+        });
```

Comment:
hey @the-other-tim-brown : where is the check for "hoodie.write.set.null.for.missing.columns". I don't see it being used or checked within AvroSchemaEvolutionUtils.reconcileSchema() 

I am wondering how can we be sure that all callers will be calling this method only when "hoodie.write.set.null.for.missing.columns" is set to true.

Reply:
I do see that we call this only when the config value is true within HoodieSchemaUtils. but I am trying to be future proof and avoid mis-use of the method.

Reply:
We can pass that through as a boolean, does that sound good to you?

Reply:
This actually requires updates in quite a few places, I will wait until there is alignment and then I can update

Reply:
yes, I prefer adding a boolean arg to avoid mis-use of the method.